kernel/gic/src/gic/redist_interface.rs

```diff
@@ -9,20 +9,42 @@
 //! - Enabling or disabling the forwarding of PPIs & SGIs based on their numbers
 //! - Getting or setting the priority of PPIs & SGIs based on their numbers
 
-use super::GicRegisters;
 use super::InterruptNumber;
 use super::Enabled;
 use super::Priority;
+use super::read_array_volatile;
+use super::write_array_volatile;
+
+use volatile::{Volatile, ReadOnly};
+use zerocopy::FromBytes;
+
+#[derive(FromBytes)]
+#[repr(C)]
+pub struct RedistRegsP1 {        // base offset
+    ctlr:         Volatile<u32>, // 0x00
+    _unused0:     u32,
+    ident:        ReadOnly<u64>, // 0x08
+    _unused1:     u32,
+    waker:        Volatile<u32>, // 0x14
+}
+
+#[derive(FromBytes)]
+#[repr(C)]
+pub struct RedistRegsSgiPpi {            // base offset
+    _reserved0:   [u8;            0x80],
+
+    group:        [Volatile<u32>; 0x01], // 0x080
+    _reserved1:   [u32;           0x1f],
+
+    set_enable:   [Volatile<u32>; 0x01], // 0x100
+    _reserved2:   [u32;           0x1f],
+
+    clear_enable: [Volatile<u32>; 0x01], // 0x180
```

Comment:
for these 1-element `Volatile<u32>` fields, what's the point of doing it like that instead of just:
```rust
    group: Volatile<u32>,
```

Reply:
Each (field + _reserved) pair corresponds to an array where only the first item is usable.
This is the best way I found to reflect that

Reply:
Also the existing code for set-enable and other bit arrays requires this to be accessible as a slice